napari/plugins/io.py

```diff
@@ -76,8 +83,20 @@ def read_data_with_plugins(
                 logger.error(err.format_with_contact_info())
 
 
-def write_data_with_plugins(
-    path: str, layer_data: LayerData, plugin_manager=None
+def write_layer_with_plugins(path: str, layer: Layer, plugin_manager=None):
+    # TODO: sidenote, I never liked this pattern, and it's getting less
+    # meaningful as we make more customizations of our plugin manager...
+    # but it might be necessary for tests?
+    plugin_manager = plugin_manager or napari_plugin_manager
```

Comment:
yeah, I'd like to consider whether would should just always use the `napari.plugins.plugin_manager` singleton.  but I think there's one place in tests where I build a new plugin manager